tensorboard/components/tf_paginated_view/tf-paginated-view2.html

```diff
@@ -0,0 +1,414 @@
+        <paper-button
+          on-tap="_performPreviousPage"
+          disabled$="[[!_hasPreviousPage]]"
+        >Previous page</paper-button><!--
+        --><paper-button
+          on-tap="_performNextPage"
+          disabled$="[[!_hasNextPage]]"
+        >Next page</paper-button>
+      </div>
+    </template>
+    <div id="template"><slot></slot></div>
+    <div id="items"></div>
+    <template is="dom-if" if="[[_multiplePagesExist]]">
+      <div id="controls-container">
+        <div style="display: inline-block; padding: 0 5px">
+          Page
+          <paper-input
+            id="page-input"
+            type="number"
+            no-label-float
+            min="1"
+            max="[[_pageCount]]"
+            value="[[_pageInputValue]]"
+            style="display: inline-block; width: [[_inputWidth]];"
+            on-input="_handlePageInputEvent"
+            on-change="_handlePageChangeEvent"
+            on-focus="_handlePageFocusEvent"
+            on-blur="_handlePageBlurEvent"
+          ></paper-input>
+          of [[_pageCount]]
+        </div>
+      </div>
+      <div class="big-page-buttons" style="margin-top: 10px;">
+        <paper-button
+          on-tap="_performPreviousPage"
+          disabled$="[[!_hasPreviousPage]]"
+        >Previous page</paper-button><!--
+        --><paper-button
+          on-tap="_performNextPageAndJumpToTop"
+          disabled$="[[!_hasNextPage]]"
+        >Next page</paper-button>
+      </div>
+    </template>
+    <style>
+      :host {
+        display: flex;
+        flex-direction: column;
+      }
+      #controls-container {
+        justify-content: center;
+        display: flex;
+        flex-direction: row;
+        flex-grow: 0;
+        flex-shrink: 0;
+        width: 100%;
+      }
+      #controls-container paper-button {
+        display: inline-block;
+      }
+      .big-page-buttons {
+        display: flex;
+      }
+      .big-page-buttons paper-button {
+        background-color: var(--tb-ui-light-accent);
+        color: var(--tb-ui-dark-accent);
+        display: inline-block;
+        flex-basis: 0;
+        flex-grow: 1;
+        flex-shrink: 1;
+        font-size: 13px;
+      }
+      .big-page-buttons paper-button[disabled] {
+        background: none;
+      }
+      #items {
+        display: flex;
+        flex-direction: row;
+        flex-wrap: wrap;
+      }
+    </style>
+  </template>
+  <script>
+  Polymer({
+    is: "tf-paginated-view2",
+    properties: {
+      /**
+       * An arbitrary list of items. Items are opaque and will not be
+       * inspected or modified.
+       */
+      items: {
+        type: Array,
+        value: () => [],
+      },
+
+      getItemKey: {
+        type: Function,
+        value: () => (item, index) => index,
+      },
+
+      as: {
+        type: String,
+        value: 'item'
+      },
+
+      /**
+       * A list of all pages, exactly one of which will be active
```

Comment:
This documentation doesn't appear to be correct any more - this looks like it's instead just the set of currently visible rendered template instances, not a list of all pages?